x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -134,6 +136,19 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
+  public async removeKeyPair(): Promise<void> {
+    const currentKeyPair = await this.getCurrentKeyPairObj();
+    if (currentKeyPair) {
+      await this.soClient.delete(MESSAGE_SIGNING_KEYS_SAVED_OBJECT_TYPE, currentKeyPair.id);
+    }
+  }
+
+  public async rotateKeyPair(): Promise<void> {
+    await this.removeKeyPair();
+    await this.generateKeyPair();
+    // TODO: Apply changes to all policies
```

Comment:
What is the impact of not applying the changes to agents using the agent policy? Do we expect any potential issues due to the changed signing key?